Assets/SEE/UI/Window/ConsoleWindow/ConsoleWindow.cs

```diff
@@ -0,0 +1,691 @@
+        private void CreatemessageItem(Message message)
+        {
+            GameObject item = PrefabInstantiator.InstantiatePrefab(itemPrefab, items, false);
+
+            Channel channel = channels.GetValueOrDefault(message.Channel);
+            Color color = channel?.Levels.GetValueOrDefault(message.Level)?.Color ?? Color.white;
+            char icon = channel?.Icon ?? '\u003f';
+
+            TextMeshProUGUI textMesh = item.transform.Find("Foreground/Text").gameObject.MustGetComponent<TextMeshProUGUI>();
+            textMesh.text = message.Text;
+            textMesh.color = color.IdealTextColor();
+
+            TextMeshProUGUI iconMesh = item.transform.Find("Foreground/Type Icon").gameObject.MustGetComponent<TextMeshProUGUI>();
+            iconMesh.text = icon.ToString();
+            iconMesh.color = color.IdealTextColor();
+
+            item.transform.Find("Background").GetComponent<UIGradient>().EffectGradient.SetKeys(
+                new Color[] { color, color.Darker(0.3f) }.ToGradientColorKeys().ToArray(), 
+                new GradientAlphaKey[] { new(1, 0), new(1, 1) });
+
+            UpdateFilter(message, item);
+        }
+
+        /// <summary>
+        /// Updates a message item.
+        /// </summary>
+        /// <param name="i">The message index.</param>
+        private void UpdateItem(int i)
+        {
+            GameObject item = items.GetChild(i).gameObject;
+            Message message = messages[i];
+            TextMeshProUGUI textMesh = item.transform.Find("Foreground/Text").gameObject.MustGetComponent<TextMeshProUGUI>();
+            Debug.Log($"Update Text - {i} - {message.Text}");
+            textMesh.SetText(message.Text);
+            UpdateFilter(message, item);
+        }
+
+        /// <summary>
+        /// Updates the filters.
+        /// </summary>
+        private void UpdateFilters()
+        {
+            for (int i = 0; i < items.childCount; i++)
+            {
+                UpdateFilter(messages[i], items.GetChild(i).gameObject);
+            }
+        }
+
+        /// <summary>
+        /// Updates the filter for a message.
+        /// </summary>
+        /// <param name="message">The message.</param>
+        /// <param name="item">The message item.</param>
+        private void UpdateFilter(Message message, GameObject item)
+        {
+            item.SetActive(true);
+
+            string text = item.transform.Find("Foreground/Text").gameObject.MustGetComponent<TextMeshProUGUI>().text;
+            if (!text.Contains(searchField.text, matchCase ? 0 : StringComparison.OrdinalIgnoreCase))
+            {
+                item.SetActive(false);
+            }
+            if (fullMatch && text.Length != searchField.text.Length)
+            {
+                item.SetActive(false);
+            }
+            if (!channels[message.Channel].Levels[message.Level].enabled)
+            {
+                item.SetActive(false);
+            }
```

Comment:
These can be combined into a single if statement, since they all do the same thing, right?